File: go-sdk/pkg/models/operations/deleteroomsroomid.go
package operations

import (
	"openapi/pkg/models/shared"
)

type DeleteRoomsRoomIDPathParams struct {
	RoomID string `pathParam:"style=simple,explode=false,name=roomId"`
}

type DeleteRoomsRoomIDRequest struct {
	PathParams DeleteRoomsRoomIDPathParams
}

type DeleteRoomsRoomIDResponse struct {
	ContentType string
	Error       *shared.Error
	StatusCode  int64
}

File: typescript-sdk/src/sdk/models/operations/deleteroomsroomid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteRoomsRoomIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=roomId" })
  roomId: string;
}


export class DeleteRoomsRoomIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DeleteRoomsRoomIdPathParams;
}


export class DeleteRoomsRoomIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  statusCode: number;
}
